Src/Models/CBR.cpp

```diff
@@ -96,25 +126,35 @@ CBR::CBR(GraphicsResources* gr, const PGE::String& filename) {
             }
             for (int k = 0; k < vertexCount; k++) {
                 PGE::Vertex tempVertex;
-                tempVertex.setVector4f("position", PGE::Vector4f(reader.readVector3f(), 1.f));
+                PGE::Vector3f position = reader.readVector3f();
+                tempVertex.setVector4f("position", PGE::Vector4f(position, 1.f));
                 tempVertex.setVector3f("normal", PGE::Vector3f::ONE);
                 tempVertex.setVector2f("lmUv", reader.readVector2f());
                 tempVertex.setVector2f("diffUv", reader.readVector2f());
                 tempVertex.setColor("color", PGE::Color::WHITE);
                 vertices[textureID].push_back(tempVertex);
+
+                vertexPositions[textureID].push_back(position);
             }
         }
     }
+
+
     meshes.reserve(texSize);
     for (int i = 0; i < texSize; i++) {
         if (toolTextures.find(i) == toolTextures.end()) {
             PGE::Mesh* newMesh = PGE::Mesh::create(gr->getGraphics(), PGE::Primitive::Type::TRIANGLE);
             newMesh->setMaterial(materials[i]);
             newMesh->setGeometry(vertices[i].size(), vertices[i], primitives[i].size(), primitives[i]);
             meshes.push_back(newMesh);
+
+            collisionMeshes.push_back(new CollisionMesh(vertexPositions[i], primitives[i]));
         }
     }
 
+
+
+    //delete[] indices;
```

Comment:
A vector can't be deleted.

Reply:
Good to know.
I need to learn how to use delete correctly.

Reply:
Everything you create with the `new` keyword has to be `delete`d, everything you create with the `new[...]` keyword has to be `delete[]`d